### PR TITLE
SOLR-16586: Log start of SearchHandler#handleRequestBody

### PR DESCRIPTION
**Pull request by patsonluk**
https://issues.apache.org/jira/browse/SOLR-16586

# Description

Add logging to start of `SearchHandler#handleRequestBody` so if processing hangs we know which query is causing that

# Solution

Added info logging to the beginning of `SearchHandler#handleRequestBody` with the query params

# Tests

No test added

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by risdenk — CHANGES_REQUESTED** [review]

File: solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java [hunk 1]
```diff
@@ -326,6 +326,9 @@ protected ResponseBuilder newResponseBuilder(
 
   @Override
   public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throws Exception {
+    if (log.isInfoEnabled()) {
+      log.info("Start Search Query: {}", req.getParamString());
+    }
```
[the remark this answers is not in the document]
Comment (patsonluk) [reply solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java:331]: Thank you for the review! I do understand your concern of extra log info, it's important for our use case as queries can sometimes be totally stuck hence timeout wouldn't get printed.

I do agree that it's probably quite specific to our own scenarios (and we might not really seeing issues like that anymore), perhaps changing it to debug will be more suitable?

[the remark this answers is not in the document]
Comment (risdenk) [reply solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java]: >  it's important for our use case as queries can sometimes be totally stuck hence timeout wouldn't get printed.

yea I totally understand that. Its always nice to have logging but at the same time this subtle change has the ability to blow up existing installs. debug is a decent idea. I thought I saw a jira/pr go around about adding markers to control logging? maybe that makes sense here to try?

I just think we need to be careful about default logging just surprising folks operating Solr.

[the remark this answers is not in the document]
Comment (patsonluk) [reply solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java]: Yes! this is the marker PR #1229 , could definitely use some review on that one too 🙇🏼 

As for this one, perhaps we can keep it simple and change it to debug instead? 😊

**Comment by magibney** [comment]
Doesn't [this](https://github.com/apache/solr/blob/6869ec97c7a59cc6acd001eb592b98ea90679fb8/solr/core/src/java/org/apache/solr/core/SolrCore.java#L2861-L2864) allow to achieve essentially the same thing (pre-logging requests), by configuring logging to:

```
org.apache.solr.core.SolrCore.Request=DEBUG
```

**Comment by patsonluk** [comment]
> Doesn't [this](https://github.com/apache/solr/blob/6869ec97c7a59cc6acd001eb592b98ea90679fb8/solr/core/src/java/org/apache/solr/core/SolrCore.java#L2861-L2864) allow to achieve essentially the same thing (pre-logging requests), by configuring logging to:
> 
> ```
> org.apache.solr.core.SolrCore.Request=DEBUG
> ```

That's a good point @magibney ! I guess for our own usage, we were having issues with Search queries in particular and didn't want to enable debug for SolrCore for that, which could create quite a bit of noise. 

However, from general Solr perspective, it does not bring much values to have 2 different debug pieces with very similar info. The scope difference - `SolrCore` vs `SearchHandler` probably does not warrant the extra change. I will keep the change to our own local version, and retract this PR for now.

Thank you for all the comments and help!

**Pull request closed without merging**